tests/core/policies/test_rule_policy.py

```diff
@@ -38,29 +42,68 @@
     LoopInterrupted,
     FollowupAction,
 )
-from rasa.shared.nlu.interpreter import RegexInterpreter
 from rasa.core.nlg import TemplatedNaturalLanguageGenerator
-from rasa.core.policies.rule_policy import RulePolicy, InvalidRule, RULES
+from rasa.core.policies.rule_policy import (
+    RulePolicyGraphComponent as RulePolicy,
+    InvalidRule,
+    RULES,
+)
+from rasa.graph_components.providers.rule_only_provider import RuleOnlyDataProvider
 from rasa.shared.core.trackers import DialogueStateTracker
 from rasa.shared.core.generator import TrackerWithCachedStates
 from tests.core import test_utils
 
 UTTER_GREET_ACTION = "utter_greet"
 GREET_INTENT_NAME = "greet"
-GREET_RULE = DialogueStateTracker.from_events(
-    "greet rule",
-    evts=[
-        ActionExecuted(RULE_SNIPPET_ACTION_NAME),
-        ActionExecuted(ACTION_LISTEN_NAME),
-        # Greet is a FAQ here and gets triggered in any context
-        UserUttered(intent={"name": GREET_INTENT_NAME}),
-        ActionExecuted(UTTER_GREET_ACTION),
-        ActionExecuted(ACTION_LISTEN_NAME),
-    ],
+GREET_RULE: TrackerWithCachedStates = cast(
+    TrackerWithCachedStates,
+    DialogueStateTracker.from_events(
+        "greet rule",
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            # Greet is a FAQ here and gets triggered in any context
+            UserUttered(intent={"name": GREET_INTENT_NAME}),
+            ActionExecuted(UTTER_GREET_ACTION),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ],
+    ),
 )
 GREET_RULE.is_rule_tracker = True
 
 
+@pytest.fixture()
+def resource() -> Resource:
+    return Resource("rule_policy")
+
+
+@pytest.fixture()
+def policy_with_config(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    resource: Resource,
+) -> Callable[..., RulePolicy]:
+    def inner(config: Optional[Dict[Text, Any]] = None) -> RulePolicy:
+        config = config or {}
+        policy = RulePolicy.create(
+            {**RulePolicy.get_default_config(), **config},
```

Comment:
I've eventually put that into the constructor (or constructor of base class) to avoid having to thing about that in tests later. Seems odd to me that we have the `get_default_config` but the default config is not enforced by the constructor.

Reply:
That's done[ by the `GraphNode`. ](https://github.com/RasaHQ/rasa/blob/5bb331ff7ba75ad564478cbc71fd1a46bce5ee02/rasa/engine/graph.py#L316). What do you mean by enforcing it?

Reply:
I mean I can instantiate components and then get an instance that has none of the default parameters set that I expect - unless we consistently use `self._config.get('whatever', self.get_default_config()['whatever'])`.

Reply:
yep, but that's only in the unit tests, isn't it? Can't happen in real-life cause of the `GraphNode`

Reply:
True - still would expect this to happen inside the graph component - same with the sanity checks for the configs (i.e. if you pass in a parameter that isn't there, the component should be able to complain)

Reply:
> the component should be able to complain

I'd still can do this (several components actually do)

> True - still would expect this to happen inside the graph component

What are the advantages for this? In my opinion it helps to remove duplicate logic in the components (overriding with default, storing config)